### PR TITLE
doc/rados: rewrite EC intro

### PR DESCRIPTION
**Pull request by zdover23**
This PR rewrites the front matter in the "Erasure Code"
section of the RADOS documentation. Previously, the information
in this section was syntactically confused. I have also fleshed
out the distinction between erasure coding and replication.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/operations/erasure-code.rst [hunk 1]
```diff
@@ -4,12 +4,25 @@
...
-meaning every object is copied on multiple disks. The `Erasure Code
-<https://en.wikipedia.org/wiki/Erasure_code>`_ pool type can be used
-instead to save space.
+By default, Ceph `pools <../pools>`_ are created with the type "replicated". In
+replicated-type pools, every object is copied to multiple disks (this
+multiple copying is the "replication").
+
+But **there is another type of Ceph pool: the** `erasure-coded <https://en.wikipedia.org/wiki/Erasure_code>`_ **pool**.
+
+Erasure-coded pools use a method of data protection that is different
```
Comment (cbodley) [inline doc/rados/operations/erasure-code.rst:13]: the bold might be a little much? what do you think about dropping that whole sentence, and just starting the next paragraph with `In contrast,`?

Comment (zdover23) [reply doc/rados/operations/erasure-code.rst:13]: I've been chewing on it, and I think that there's value in introducing the concept of erasure coded pools in this kind of bold way *if and only if this isn't meant to be reference material*. This material is potentially the first time that the reader is learning about erasure coding (I know that this is not true of you, and it is not true of all the other storage engineers who eat erasure coding for breakfast, but you guys don't need a sentence like this).

This is (I think) our old discussion about the target audience of the documentation. I'm thinking about the 15-year-old kid who wants to set up a Ceph cluster and has never imagined anything fancier than RAID as well as the 47-year-old data center maintainer who knows the whole history of erasure coding back to Bell Labs. I think that the absolute value of the information for the 15-year-old kid in this kind of declarative paragraph outweighs the absolute value of the old-hat impertinence of the paragraph for the 47-year-old storage wizard.

If you push back, I'll concede in order to get (most of) this change into the docs, but I thought that I would press my case one more time.

Comment (cbodley) [reply doc/rados/operations/erasure-code.rst:13]: the bolded sentence just feels click-baity to me

> This material is potentially the first time that the reader is learning about erasure coding 

i'm still struggling with this rationale. if the user doesn't know what erasure coding is, why does ceph need to push it as a solution?

replicated pools are the default for a reason. i'm no EC expert myself, i just know that there are tradeoffs (extra cpu usage for extra capacity) that mean it's only suitable for some deployments

**Review by cbodley — COMMENTED** [review]
@jdurgin @neha-ojha kindly requesting that you guys take over this review

File: doc/rados/operations/erasure-code.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/operations/erasure-code.rst]: Casey,

I said that I'd concede if you insisted, and I have.

I hadn't seen the sentence as click-baity before you mentioned it, but I see your point.

I've made the change you requested.

**Comment by zdover23** [comment]
Superseded by https://github.com/ceph/ceph/pull/48322.

**Pull request closed without merging**